src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -788,6 +792,18 @@ fn file_metadata_raw(
 
             let file_name = file_name.as_deref().unwrap_or("<unknown>");
             let directory = directory.as_deref().unwrap_or("");
+            let checksum_string;
+            let (checksum_kind, checksum_value) = match checksum {
+                Some(checksum) => {
+                    let kind = match checksum.kind {
+                        rustc_span::SourceFileChecksumKind::Md5 => llvm::ChecksumKind::MD5,
+                        rustc_span::SourceFileChecksumKind::Sha1 => llvm::ChecksumKind::SHA1,
+                    };
+                    checksum_string = hex::encode(checksum.bytes());
+                    (kind, checksum_string.as_str())
+                }
+                None => (llvm::ChecksumKind::None, ""),
```

Comment:
You can replace the second value of the tuple with `String::new` (which doesn't allocate):
```rust
let (checksum_kind, checksum_value) = match checksum {
  Some(checksum) => {
      let kind = match checksum.kind {
          rustc_span::SourceFileChecksumKind::Md5 => llvm::ChecksumKind::MD5,
          rustc_span::SourceFileChecksumKind::Sha1 => llvm::ChecksumKind::SHA1,
       };
  
      (kind, hex::encode(checksum.bytes()))
   }
   None => (llvm::ChecksumKind::None, String::new()),
}
```

Reply:
Thanks. That's much nicer.